bigtable-hbase-1.x-parent/bigtable-hbase-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java

```diff
@@ -348,23 +363,163 @@ public HBaseSchemaTranslator(SchemaTranslationOptions options) throws IOExceptio
+        throws IOException, DeserializationException {
+      ClusterSchemaDefinition transformedSchema = new ClusterSchemaDefinition();
+
+      // Parse the mappings
+      parseMappingFile();
+
+      // Apply the transformations.
+      for (TableSchemaDefinition tableSchemaDefinition : originalSchema.tableSchemaDefinitions) {
+        String tableName = tableSchemaDefinition.name;
+        HTableDescriptor tableDescriptor = tableSchemaDefinition.getHbaseTableDescriptor();
+
+        // Override the table name if its present in the mapping file
+        if (tableNameMappings.containsKey(tableName)) {
+          tableName = tableNameMappings.get(tableName);
+        }
+        // Rename the table and copy all the other configs, including the column families.
+        HTableDescriptor newTableDescriptor =
+            new HTableDescriptor(TableName.valueOf(tableName), tableDescriptor);
+
+        // For all the column families that need renaming, replace them.
+        if (tableScopedColumnFamilyMappings.containsKey(tableSchemaDefinition.name)) {
+          Map<String, String> cfMap =
+              tableScopedColumnFamilyMappings.get(tableSchemaDefinition.name);
+          for (Map.Entry<String, String> cfEntry : cfMap.entrySet()) {
+            // Its not possible to rename a columnFamily using the HColumnDescriptor interface
+            // So create a new family with same config and replace it in the HTableDescriptor.
+            newTableDescriptor.addFamily(
+                renameFamily(
+                    cfEntry.getValue(), newTableDescriptor.getFamily(cfEntry.getKey().getBytes())));
+            newTableDescriptor.removeFamily(cfEntry.getKey().getBytes());
+          }
+        } else {
+          // All the column families are copied already, nothing else to do.
+        }
+        transformedSchema.addTableSchemaDefinition(
+            newTableDescriptor, tableSchemaDefinition.splits);
+      }
+      return transformedSchema;
+    }
+
+    private HColumnDescriptor renameFamily(
+        final String newFamilyName, final HColumnDescriptor columnDescriptor) {
+      HColumnDescriptor newColumnDescriptor = new HColumnDescriptor(newFamilyName);
+      for (Map.Entry<ImmutableBytesWritable, ImmutableBytesWritable> valueEntry :
+          columnDescriptor.getValues().entrySet()) {
+        newColumnDescriptor.setValue(
+            valueEntry.getKey().copyBytes(), valueEntry.getValue().copyBytes());
+      }
+      for (Map.Entry<String, String> configEntry : columnDescriptor.getConfiguration().entrySet()) {
+        newColumnDescriptor.setConfiguration(configEntry.getKey(), configEntry.getValue());
+      }
+      return newColumnDescriptor;
+    }
+  }
+
   @VisibleForTesting
-  HBaseSchemaTranslator(SchemaReader schemaReader, SchemaWriter schemaWriter) {
+  HBaseSchemaTranslator(
+      SchemaReader schemaReader, SchemaWriter schemaWriter, SchemaTransformer schemaTransformer) {
```

Comment:
consider keeping the old ctor and have it delegate to the new one with a NoopTransformer...it should minimize the refactor noise